app/src/test/kotlin/com/hedvig/app/feature/genericauth/OtpInputViewModelTest.kt

```diff
@@ -201,18 +168,60 @@ class OtpInputViewModelTest {
 
   @Test
   fun `updating the input after getting an error should clear the error`() = runTest {
-    otpResult = SubmitOtpResult.Error("Error")
-    viewModel.setInput("111111")
+    val authRepository = FakeAuthRepository()
+    val viewModel = testViewModel(authRepository)
+
     viewModel.submitCode("111111")
+    runCurrent()
     assertThat(viewModel.viewState.value.loadingCode).isEqualTo(true)
     assertThat(viewModel.viewState.value.loadingResend).isEqualTo(false)
     assertThat(viewModel.viewState.value.networkErrorMessage).isNull()
 
-    advanceUntilIdle()
+    authRepository.submitOtpResponse.add(SubmitOtpResult.Error(""))
+    runCurrent()
     assertThat(viewModel.viewState.value.loadingCode).isEqualTo(false)
     assertThat(viewModel.viewState.value.networkErrorMessage).isNotNull()
 
     viewModel.setInput("1")
     assertThat(viewModel.viewState.value.networkErrorMessage).isNull()
   }
+
+  private fun TestScope.testViewModel(
+    authRepository: FakeAuthRepository,
+    authTokenStorage: AuthTokenStorage = testAuthTokenStorage(),
+  ): OtpInputViewModel {
+    return OtpInputViewModel(
+      verifyUrl = "verifytest",
+      resendUrl = "resendtest",
+      credential = "test@email.com",
+      authTokenService = testAuthTokenService(authRepository, authTokenStorage),
+      authRepository = authRepository,
+      uploadMarketAndLanguagePreferencesUseCase = mockk(relaxed = true),
+    )
+  }
+
+  private fun TestScope.testAuthTokenStorage(): AuthTokenStorage {
+    return AuthTokenStorage(
+      dataStore = TestPreferencesDataStore(
+        datastoreTestFileDirectory = testFolder.newFolder("datastoreTempFolder"),
+        backgroundScope,
```

Comment:
backgroundScope is a field inside `TestScope` which we get inside the `runTest{}` block. It serves as a CoroutineScope which can be used to run stuff which is okay to still be running at the end of the test without making the test hang.
This is needed since for this example running a DataStore needs a CoroutineScope but it'd stay open forever and make the test timeout otherwise.
This lets it run in the background using our test CoroutineScope

Discussion where I learned this here https://slack-chats.kotlinlang.org/t/8113862/i-got-a-coroutine-test-where-i-use-the-testscope-provided-in